### PR TITLE
Change ISBN search URL

### PR DESCRIPTION
**Pull request by adam3smith**
make sure it detects for both individual and multiple search results. Possibly fixes, but definitely improves, https://forums.zotero.org/discussion/64880/isbn-problem

### CODE REVIEW

**Comment by adam3smith** [comment]
@zuphilip if you could take a look when you have a little time?

File: Open WorldCat.js [hunk 1]
```diff
@@ -341,7 +340,17 @@ function fetchIDs(isbns, ids, callback) {
 				} else {
 					Z.debug("Could not extract OCLC ID for ISBN " + isbn);
 				}
-			} else {
+			}
+			//but sometimes we have single items
+			else if (getFirstContextObj(doc)) {
```
Comment (zuphilip) [inline Open WorldCat.js]: The `getFirstContextObj` is AFAIK just checking for the presence of COinS data, but this is also available in search pages, e.g. http://www.worldcat.org/search?qt=worldcat_org_all&q=mannheim

File: Open WorldCat.js [hunk 2]
```diff
@@ -341,7 +340,17 @@ function fetchIDs(isbns, ids, callback) {
...
+				var canonicalURL = ZU.xpath(doc, '/html/head/link[@rel="canonical"][1]')[0];
```
Comment (zuphilip) [inline Open WorldCat.js]: I would suggest:
```js
var canonicalURL = ZU.xpathText(doc, '/html/head/link[@rel="canonical"]/@href');
```
(It looks IMO not likely that a website has several canonical links.)

File: Open WorldCat.js (hunk 1 above)
Comment (zuphilip) [inline Open WorldCat.js]: Isn't it here enough to check for the presence of this canonical url, i.e. something like
```js
} else {
   var canonicalURL = ZU.xpathText(doc, '/html/head/link[@rel="canonical"]/@href');
   if (canonicalURL) {
      oclcID = extractOCLCID(canonicalURL);
      if(!oclcID) throw new Error("WorldCat: Failed to extract OCLC ID from URL: " + url);
         scrape([oclcID]);
      }
   } else {
      Z.debug("No search results found for ISBN " + isbn);
   }
}
```
?

File: Open WorldCat.js [hunk 3]
```diff
@@ -327,11 +326,11 @@ function fetchIDs(isbns, ids, callback) {
 	}
 	
 	var isbn = isbns.shift();
-	var url = "http://www.worldcat.org/search?qt=results_page&q=bn%3A"
+	var url = "http://www.worldcat.org/search?qt=results_page&q=isbn%3A"
```
Comment (zuphilip) [inline Open WorldCat.js]: I assume that you tested this change.

Comment (adam3smith) [inline Open WorldCat.js]: yes, that's tested. It behaves slightly

**Pull request merged**